tools/azsdk-cli/Azure.Sdk.Tools.Cli.Analyzer/MCPAnalyzer.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.Sdk.Tools.Cli.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MCPAnalyzer : DiagnosticAnalyzer
+    {
+        public const string Id = "MCP001";
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            Id,
+            "McpServerTool methods must wrap body in try/catch, see the README within the tools directory for examples",
+            "Method '{0}' must have its entire body inside try{{}}catch(Exception)",
+            "Reliability",
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext ctx)
+        {
+            ctx.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            ctx.EnableConcurrentExecution();
+            ctx.RegisterSyntaxNodeAction(AnalyzeMethod, SyntaxKind.MethodDeclaration);
+        }
+
+        /// <summary>
+        /// This is the main bulk of this analyzer. It scans for methods that are decorated with MCPServerTool attribute,
+        /// and ensures those checks properly wrap their body in a try/catch block. This catch should be a generalized System.Exception.
+        /// Within the catch block, there must be a SetFailure() call.
```

Comment:
before I dig further and do this second case, I wanted to get opinions. I've only spun on this PR for ~1.5 hours all told so it's not a huge waste of my time so far.